Dockerfile

```diff
@@ -1,34 +1,79 @@
-ARG REGISTRY=docker.io/stsdockerhub
-ARG LARAVEL_ALPINE_VERSION=8.3.2-laravel-alpine3.19
-
-FROM ${REGISTRY}/php:${LARAVEL_ALPINE_VERSION}-build as build-container
-
-RUN docker-php-ext-install mysqli pdo pdo_mysql && docker-php-ext-enable pdo_mysql
-
+FROM composer:2.2 AS composer_base
+
+#-----------------------------------------------------------------
+FROM alpine:3.20 AS base
+
+ARG REPO=https://github.com/cioraneanu/firefly-pico.git
+ARG BRANCH=main
+
+
+#Install packages
+RUN apk add --no-cache \
+    tar \
+    nginx \
+    nodejs \
+    npm \
+    supervisor \
+    php82 \
+    php82-session \
+    php82-ctype \
+    php82-fpm \
+    php82-pdo \
+    php82-opcache \
+    php82-zip \
+    php82-phar \
+    php82-iconv \
+    php82-cli \
+    php82-curl \
+    php82-openssl \
+    php82-mbstring \
+    php82-tokenizer \
+    php82-fileinfo \
+    php82-json \
+    php82-xml \
+    php82-xmlwriter \
+    php82-simplexml \
+    php82-dom \
+    php82-pdo_mysql \
+    php82-pdo_pgsql \
+    php82-tokenizer
+RUN ln -s /usr/bin/php82 /usr/bin/php
+
+#-----------------------------------------------------------------
+FROM base AS build-container
+
+#Install packages
+RUN apk add --no-cache git
+
+# Installing composer
+COPY --from=composer_base /usr/bin/composer /usr/local/bin/composer
+
+#Pull Firefly-Pico from repo
+WORKDIR /var/www/html
+RUN git init \
```

Comment:
Any reason to clone from git instead of COPY cmd?

Reply:
Not really. I just lazy copied that over.

Somewhat of an argument is that you no can actually build it w/o downloading the repo. This would also allow you to skip the checkout step in github workflow. In the end saves some cpu/bandwith.